test/expected/hnsw_cost_estimate.out

```diff
@@ -1,116 +1,129 @@
 SET client_min_messages=debug5;
+\ir utils/sift10k_array.sql
+CREATE TABLE IF NOT EXISTS sift_base10k (
+     id SERIAL PRIMARY KEY,
+     v REAL[128]
+);
+\copy sift_base10k (v) FROM '/tmp/lantern/vector_datasets/siftsmall_base_arrays.csv' with csv;
+-- This function executes the given SQL query and returns its estimated total cost.
+-- It parses the EXPLAIN output to retrieve the outermost (top-level) cost estimation.
+-- Returns NULL if no cost is found or if the provided query doesn't match the expected format.
+CREATE OR REPLACE FUNCTION get_cost_estimate(explain_query text) RETURNS real AS $$
+DECLARE
+    explain_output text;
+    cost_match text;
+    line text;
+BEGIN
+    EXECUTE explain_query INTO explain_output;
+    FOR line IN (
+        SELECT
+            unnest(string_to_array(explain_output, E'\n')))
+        LOOP
+            IF position(' ' IN
+            LEFT (line, 1)) = 0 AND line LIKE '%cost=%' THEN
+                cost_match :=(regexp_matches(line, 'cost=\d+\.\d+..\d+\.\d+'))[1];
```

Comment:
I find it easier to read regex when there is an example string you are trying to match in a comment.
Maybe add an example cost estimate and show how these lines break it down?